pkg/datapath/garp/garp.go

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package garp
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/netip"
+
+	"github.com/mdlayher/arp"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+
+	"github.com/cilium/cilium/pkg/hive/cell"
+)
+
+const (
+	// GARPInterface is the interface used to send Gratuitous ARP messages.
+	GARPInterface = "garp-interface"
+)
+
+var Cell = cell.Module(
+	"garp",
+	"GARP",
+
+	cell.Provide(newGARPSender),
+
+	// This cell can't have a default config, it's entirely env dependent.
+	cell.Config(Config{}),
+)
+
+// Config contains the configuration for the GARP cell.
+type Config struct {
+	GARPInterface string
+}
+
+func (def Config) Flags(flags *pflag.FlagSet) {
+	flags.String(GARPInterface, def.GARPInterface, "Interface used for sending gratuitous arp messages")
+}
+
+type Sender interface {
+	Send(netip.Addr) error
```

Comment:
Could you add a second function which allows the caller to specify the network interface that is used. For L2-Aware LB I may need to GARP via multiple network devices which are defined via custom resources not cilium config.

Reply:
Done. I've made it a method so you get free logging, but I'm happy to just make it a normal function if that's more preferable.